File: C#/URI/1040.cs

using System;
using System.Globalization;

namespace uri
{
    class Program
    {

        static void Main(string[] args)
        {

            float n1, n2, n3, n4, media;
            float exameFinal = 0;
            float mediaFinal = 0;

            string[] vet = Console.ReadLine().Split(' ');
            n1 = float.Parse(vet[0], CultureInfo.InvariantCulture);
            n2 = float.Parse(vet[1], CultureInfo.InvariantCulture);
            n3 = float.Parse(vet[2], CultureInfo.InvariantCulture);
            n4 = float.Parse(vet[3], CultureInfo.InvariantCulture);

            media = (n1 * 2 + n2 * 3 + n3 * 4 + n4 * 1) / 10;
            media = (float)Math.Truncate(10 * media) / 10;

            if (media < 7.0 && media >= 5.0)
            {
                exameFinal = float.Parse(Console.ReadLine(), CultureInfo.InvariantCulture);

                mediaFinal = (media + exameFinal) / 2;

            }

            if (media >= 7.0 || mediaFinal >= 5.0)
            {
                Console.WriteLine("Media: " + media.ToString("F1", CultureInfo.InvariantCulture));

                if (media < 7.0 || mediaFinal >= 5.0)
                {
                    Console.WriteLine("Aluno em exame.");
                    Console.WriteLine("Nota do exame: " + exameFinal.ToString("F1", CultureInfo.InvariantCulture));
                    Console.WriteLine("Aluno aprovado.");
                    Console.WriteLine("Media final: " + mediaFinal.ToString("F1", CultureInfo.InvariantCulture));
                }

                else if (media >= 7.0)
                {
                    Console.WriteLine("Aluno aprovado.");
                }

                else
                {
                    Console.WriteLine("Aluno reprovado.");
                }
            }
            else
            {
                Console.WriteLine("Media: " + media.ToString("F1", CultureInfo.InvariantCulture));
                Console.WriteLine("Aluno reprovado.");
            }
        }
    }
}

File: C/URI/1040.c
#include <stdio.h>
 
int main() {
 
  double n1, n2, n3, n4, media, exame, mediaexame;
        scanf("%lf %lf %lf %lf", &n1, &n2, &n3, &n4);
        media = ((n1*2)+(n2*3)+(n3*4)+(n4*1))/10;
        if(media >= 7)
                printf("Media: %.1lf\nAluno aprovado.\n", media);
        else if(media >=5 && media <= 6.9)
        {

                        printf("Media: %.1lf\n", media);
                        printf("Aluno em exame.\n");


                scanf("%lf", &exame);
                mediaexame = (media + exame)/2;
                if(mediaexame >= 5)
                {
                        printf("Nota do exame: %.1lf\n", exame);
                        printf("Aluno aprovado.\n");
                        printf("Media final: %.1lf\n", mediaexame);
                }
                else 
                {
                        printf("Nota do exame: %.1lf\n", exame);
                        printf("Aluno reprovado.\n");
                        printf("Media final: %.1lf\n", mediaexame);

                }
        }
        else if(media < 5)
                printf("Media: %.1lf\nAluno reprovado.\n", media);


 
    return 0;
}